### ISSUE TITLE
Prisma not compiling with entities named "Foo" and "FooUpdate"

### ISSUE THREAD

**Issue by kapv89** [prisma/prisma#9421]
### Bug description

When I try to compile my project which uses prisma, I get the following error

```

> api@1.0.0 build
> npm run clean && tsc && npm run prisma generate


> api@1.0.0 clean
> rm -rf dist

node_modules/.prisma/client/index.d.ts(4002,43): error TS2344: Type 'S["include"][P]' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'S["include"]["updates"]' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'S["include"]["updates"]' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
      Type 'S["include"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
        Type 'S["include"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
          Type 'S["include"][P]' is not assignable to type 'DiagramUpdateArgs'.
            Type 'S["include"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
              Type 'S["include"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4013,43): error TS2344: Type 'S["select"][P]' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'S["select"]["updates"]' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'S["select"]["updates"]' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
      Type 'S["select"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
        Type 'S["select"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
          Type 'S["select"][P]' is not assignable to type 'DiagramUpdateArgs'.
            Type 'S["select"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
              Type 'S["select"]["updates"]' is not assignable to type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4357,197): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateFindManyArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateFindManyArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'data' is missing in type 'DiagramUpdateFindManyArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4558,15): error TS2300: Duplicate identifier 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4878,218): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateFindUniqueArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateFindUniqueArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'data' is missing in type 'DiagramUpdateFindUniqueArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4878,344): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
node_modules/.prisma/client/index.d.ts(4895,217): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateFindFirstArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateFindFirstArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'data' is missing in type 'DiagramUpdateFindFirstArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4895,343): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
node_modules/.prisma/client/index.d.ts(4915,104): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateFindManyArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'T' is not assignable to type 'DiagramUpdateArgs'.
      Type 'DiagramUpdateFindManyArgs' is not assignable to type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4931,119): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateCreateArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateCreateArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'where' is missing in type 'DiagramUpdateCreateArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4963,119): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateDeleteArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateDeleteArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'data' is missing in type 'DiagramUpdateDeleteArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(4982,119): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateUpdateArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateUpdateArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Type 'DiagramUpdateUpdateArgs' is not assignable to type 'DiagramUpdateArgs'.
          Types of property 'include' are incompatible.
            Type 'DiagramUpdateInclude | null | undefined' is not assignable to type 'DiagramInclude | null | undefined'.
              Type 'DiagramUpdateInclude' has no properties in common with type 'DiagramInclude'.
node_modules/.prisma/client/index.d.ts(5040,119): error TS2344: Type 'T' does not satisfy the constraint 'boolean | DiagramUpdateArgs | null | undefined'.
  Type 'DiagramUpdateUpsertArgs' is not assignable to type 'boolean | DiagramUpdateArgs | null | undefined'.
    Type 'DiagramUpdateUpsertArgs' is not assignable to type 'DiagramUpdateArgs'.
      Type 'T' is not assignable to type 'DiagramUpdateArgs'.
        Property 'data' is missing in type 'DiagramUpdateUpsertArgs' but required in type 'DiagramUpdateArgs'.
node_modules/.prisma/client/index.d.ts(5491,15): error TS2300: Duplicate identifier 'DiagramUpdateArgs'.
```

### How to reproduce

1. Use the schema provided in the issue
2. Run `prisma migrate dev` or `prisma db push`
3. Try to build the project by running `tsc`

### Expected behavior

Project should compile successfully

### Prisma information

schema
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id String @db.Uuid @default(uuid()) @id
  name String
  email String @unique
  hashedPassword String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  seq Int @default(autoincrement())

  teamMembers TeamMember[]
  diagramUpdates DiagramUpdate[]

  @@index([seq])
}

model Team {
  id String @db.Uuid @default(uuid()) @id
  name String @unique
  isFree Boolean
  numSeats Int
  numUsers Int
  numDiagrams Int
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  seq Int @default(autoincrement())

  members TeamMember[]
  diagrams Diagram[]

  @@index([seq])
}

model TeamMember {
  id String @db.Uuid @default(uuid()) @id
  email String?
  userId String? @db.Uuid
  user User? @relation(fields: [userId], references: [id])
  teamId String @db.Uuid
  team Team @relation(fields: [teamId], references: [id])
  role TeamMemberRole
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  seq Int @default(autoincrement())

  @@index([seq])
  @@unique([userId, teamId])
  @@unique([email, teamId])
}

enum TeamMemberRole {
  member
  manager
  admin
}

model Diagram {
  id String @db.Uuid @default(uuid()) @id
  teamId String @db.Uuid
  team Team @relation(fields: [teamId], references: [id])
  name String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  seq Int @default(autoincrement())

  updates DiagramUpdate[]

  @@index([seq])
  @@unique([teamId, name])
}

model DiagramUpdate {
  id String @db.Uuid @default(uuid()) @id
  diagramId String @db.Uuid
  diagram Diagram @relation(fields: [diagramId], references: [id])
  userId String @db.Uuid
  user User @relation(fields: [userId], references: [id])
  data String
}
```

### Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Node.js version: v14.17.5

### Prisma Version

```
prisma                  : 3.1.1
@prisma/client          : 3.1.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine c22652b7e418506fab23052d569b85d3aec4883f (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli c22652b7e418506fab23052d569b85d3aec4883f (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine    : introspection-core c22652b7e418506fab23052d569b85d3aec4883f (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary           : prisma-fmt c22652b7e418506fab23052d569b85d3aec4883f (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash    : c22652b7e418506fab23052d569b85d3aec4883f
Studio                  : 0.423.0
```

**Comment by janpio** [comment]
> 7.22.0

There is no Prisma `7.22.0`. Can you please make sure the version information is correct? Thanks.

**Comment by kapv89** [comment]
@janpio updated the opening post in this issue with the correct prisma version info. I have prisma under an npm script in my project for dotenv related config and it was giving that version

**Comment by kapv89** [comment]
@janpio realised this issue is due to an entity named `Diagram` and another entity named `DiagramUpdate`. changing the name of `DiagramUpdate` to something else fixes the issue.

**Comment by janpio** [comment]
Can you open an issue about this specifically? That certainly should not happen and we definitely want to fix this @kapv89!

But great you figured it out and were unblocked :+1:

**Comment by kapv89** [comment]
@janpio i am kind of short of time. I'll simply reopen this issue.

**Comment by pantharshit00** [comment]
I opened a new issue for the problem with a reproduction: https://github.com/prisma/prisma/issues/9568
